Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -182,7 +183,11 @@ void UPSHUDComponent::DisplayLevelUIOverlay(bool IsLevelLocked)
 		return;
 	}
 
-	constexpr bool bShouldPlayFadeAnimation = true;
+	USettingsWidget* SettingsWidget = UMyBlueprintFunctionLibrary::GetSettingsWidget();
+	FString InstantCharacterSwitchSetting;
+	SettingsWidget->GetSettingRow(UPSDataAsset::Get().GetInstantCharacterSwitchTag()).Checkbox.GetSettingValue(*SettingsWidget, UPSDataAsset::Get().GetInstantCharacterSwitchTag(), InstantCharacterSwitchSetting);
+
+	const bool bShouldPlayFadeAnimation = InstantCharacterSwitchSetting.Equals("false", ESearchCase::IgnoreCase);
```

Comment:
`GetSettingRow` returns internal structure. In your case, just use directly the checkbox Getter - `GetCheckboxValue`, which already returns bool. As result, all these 3 lines can be replaced by single one:
```cpp
const bool bShouldPlayFadeAnimation = !SettingsWidget->GetCheckboxValue(UPSDataAsset::Get().GetInstantCharacterSwitchTag());
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/25/commits/49d346fe50f68fac2e406eed94d87fc5d3fca5c1